addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,330 @@
+        // get the zone number in integer form
+        Integer zone = Integer.parseInt(channel.getId().substring((channel.getId().length() - 1)));
+
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+
+        // if the pin is an actuator add to actuator string
+        // else add to sensor string
+
+        if (isAct.get(zone)) {
+            actuators.add("{\"pin\":" + Integer.toString(pin) + "}");
+            StringType state = new StringType("0");
+            updateState(channel, state);
+        }
+
+        else {
+            sensors.add("{\"pin\":" + Integer.toString(pin) + "}");
+        }
+
+        logger.debug(sensors.toString());
+        // launch the update so the new channel settings can be sent
+        updateKonnectedModule();
+    }
+
+    @Override
+    public synchronized void channelUnlinked(ChannelUID channel) {
+        logger.debug("Channel {} has been unlinked", channel.toString());
+        // get the zone number in integer form
+        setisAct();
+        Integer zone = Integer.parseInt(channel.getId().substring((channel.getId().length() - 1)));
+        // convert the zone to the pin based on value at index of zone
+        Integer pin = Arrays.asList(PIN_TO_ZONE).get(zone);
+
+        actuators.remove("{\"pin\":" + Integer.toString(pin) + "}");
+
+        sensors.remove("{\"pin\":" + Integer.toString(pin) + "}");
+
+        logger.debug(sensors.toString());
+        updateKonnectedModule();
+    }
+
+    /*
+     * This method constructs the payload that will be sent
+     * to the konnected module via the put request
+     * it adds the appropriate senors and actuators
+     * as well as the location of the callback servelet
+     * and auth_token which can be used for validation
+     *
+     */
+    private String contructSettingsPayload() {
+        String hostPath = "";
+        try {
+            hostPath = getHostName() + webHookServlet.getPath();
+            logger.debug("The host path is: {}", hostPath);
+        }
+
+        catch (UnknownHostException e) {
+            logger.debug("Unable to obtain hostname: {}", e);
+            return "none";
+        }
+
+        String authToken = config.Auth_Token;
+        logger.debug("The Auth_Token is: {}", authToken);
+        logger.debug("The Sensor String is: {}", sensors.toString());
+        logger.debug("The Actuator String is: {}", actuators.toString());
+
+        updatepinstate();
+        String payload = "{\"sensors\":" + sensors.toString() + ",\"actuators\": " + actuators.toString()
+                + ",\"token\": \"" + authToken + "\",\"apiUrl\": \"http://" + hostPath + "\"}";
+        logger.debug("The payload is: {}", payload);
+        return payload;
+    }
+
+    protected String getHostName() throws UnknownHostException {
+        // returns the local address of the openHAB server
+        InetAddress addr = InetAddress.getLocalHost();
```

Comment:
Instead of getting the address directly its better to use our `NetworkAddressService`. (https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java) Let the OSGi framework inject it into your factory and pass it over to the handler in  the constructor.